b_cfn_elasticsearch_index/function.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import annotations
+
+from functools import lru_cache
+
+from aws_cdk.aws_lambda import Code, SingletonFunction, Runtime
+from aws_cdk.core import Stack, Duration
+
+from b_elasticsearch_layer.layer import Layer as ElasticsearchLayer
+
+
+class ElasticsearchIndexProviderFunction(SingletonFunction):
+    """
+    Custom activity resource Singleton Lambda function.
+
+    Creates activities on stack creation.
+    Updates activities on their name changes.
+    Deletes activities on stack deletion.
+    """
+
+    def __init__(
+            self,
+            scope: Stack,
+            name: str,
+    ) -> None:
+        self.__scope = scope
+        self.__name = name
+
+        super().__init__(
+            scope=scope,
+            id=name,
+            uuid=f'{name}-uuid',
+            function_name=name,
+            code=self.__code(),
+            handler='index.handler',
+            runtime=Runtime.PYTHON_3_8,
+            layers=[ElasticsearchLayer(scope, f'{name}ElasticsearchLayer')],
+            environment={},
+            timeout=Duration.minutes(10)
+        )
+
+    def role(self) -> Role:
+        inline_policies = {
+            "ElasticsearchIndexPrividerFunctionElasticsearchAccessPolicy": PolicyDocument(
+                statements=[
+                    PolicyStatement(
+                        actions=["es:*"],  # TODO restrict this to appropriate API calls
```

Comment:
Please dont have inline comments. Also comments should end with a dot. It is an English sentence after all.